libs/ledger-live-common/src/exchange/swap/getExchangeRates.ts

```diff
@@ -40,19 +35,21 @@ const getExchangeRates: GetExchangeRates = async (
   const amountFrom = transaction.amount;
   const tenPowMagnitude = new BigNumber(10).pow(unitFrom.magnitude);
   const apiAmount = new BigNumber(amountFrom).div(tenPowMagnitude);
+  const providers = await getProviders();
+
   const request = {
     from,
     to,
     amountFrom: apiAmount.toString(),
-    providers: usesV3 ? getAvailableProviders() : undefined,
+    providers: providers.map((item) => item.provider),
```

Comment:
The API expects a `String[]` of provider names here, I don't think that works ? `providers` contains the full map with list of pairs etc.

Also even if this was the list of providers names, this covers some cases but not all:

- provider is geoblocked => OK, works, because it will be absent from `getProviders` response
- provider does not support this specific pair but supports others => KO, it will be included when it should not

You need to check if `getProviders` contains this requests's `from, `to` to include it

Reply:
Changed.